app/assets/javascripts/covid19/src/webapp/components/structures-table/CustomCheckboxFilter.tsx

```diff
@@ -90,39 +141,75 @@ export const CustomCheckboxFilter: React.FC<CustomCheckboxFilterProps> = React.m
 });
 
 const FilterItem: React.FC<FilterItemProps> = React.memo(props => {
-    const { filterKey, filterState, setFilterState } = props;
+    const { filterKey, filterState, setFilterState, tooltip } = props;
+    const [open, setOpen] = React.useState(false);
 
-    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setFilterState({
-            ...filterState,
-            [event.target.name]: event.target.checked,
-        });
-    };
+    const handleChange = React.useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            setFilterState({
+                ...filterState,
+                [event.target.name]: event.target.checked,
+            });
+        },
+        [filterState, setFilterState]
+    );
 
-    const handleClick = (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
-        const filter = event.currentTarget.dataset.filter;
-        if (filter === undefined) return;
+    const handleClick = React.useCallback(
+        (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+            const filter = event.currentTarget.dataset.filter;
```

Comment:
Note that,  similarly to the previous note, accessing target attributes is not type-safe (for example: `event.currentTarget.dataset` has a very broad type `DOMStringMap`. In general, using data attributes in React is a bit unorthodox (subjective)